sycl/include/CL/sycl/detail/sycl_mem_obj_t.hpp

```diff
@@ -184,6 +190,21 @@ class __SYCL_EXPORT SYCLMemObjT : public SYCLMemObjI {
     };
   }
 
+  template <typename Destination>
+  __SYCL_DLL_LOCAL EnableIfOutputIteratorBool<Destination>
+  set_final_data(Destination FinalData) {
+    MUploadDataFunctor = [this, FinalData]() {
+      using DestinationValueT = iterator_value_type_t<Destination>;
+      // TODO if Destination is ContiguousIterator then don't create
+      // ContiguousStorage. updateHostMemory works only with pointer to
+      // continuous data.
+      const size_t Size = MSizeInBytes / sizeof(DestinationValueT);
+      bool *ContiguousStorage = new bool[Size];
```

Comment:
There is no matching `delete` for the array.
It's better to employ `unique_ptr` here with proper deleter set for automatic freeing of memory (either on smooth operation or on exception).

Reply:
Changed to unique_ptr